xbmc/platform/Platform.cpp

```diff
@@ -47,3 +53,22 @@ void CPlatform::Init()
   // nothing for now
 }
 
+std::string CPlatform::GetUniqueHardwareIdentifier()
+{
+  static std::string uuid = NoValidUUID;
+
+#if defined(HAS_LINUX_NETWORK) || defined(HAS_WIN32_NETWORK)
```

Comment:
i will simply remove those ifdefs - if g_application.getNetwork().GetFirstConnectedInterface(); the platform is broken - there is no platform which doesn't implement that anyway ... the ifdefs where a c&p from somewhere else - i don't see why those are required at all.